### DISCUSSION TITLE
Running import from a tool

### DISCUSSION THREAD

**Discussion by aakarim** [cue-lang/cue#1488] [Q&A]
Is it a bad idea to run cue import from within a tool command? 

I'd like to standardise the way that existing YAML files are imported per vendor. For example, with Argo Workflows if I want to import an existing workflow file I'll use ```cue import -p argo -l 'strings.TrimSuffix(metadata.generateName, "-")'``` to remove the last character and use the resulting string as the key.

A common workflow for us is to define the workflow first in YAML, because there are simply so many examples out there. Once we've made it and it's confirmed to be working we'll import it into CUE and DRY it up. However, my team may not realise that the convention is the use the `generateName` and remove the suffix. A tool seems like the right answer here.

**Comment by myitcv** [comment]
> Is it a bad idea to run cue import from within a tool command?

Assuming here you mean within a `cue cmd` command (the tooling layer)?

Or do you mean a Go-based "tool" more generally? 

I'd say either is good practice. If you are DRY-ing things up via a `cue cmd` command, then you will obviously be able to rely on `cmd/cue` being available.

If you are writing a Go-based tool, then you can import and use https://pkg.go.dev/cuelang.org/go/cmd/cue/cmd, which is the same wrapper used by `cmd/cue`.
